priiloader/source/hacks.cpp

```diff
@@ -485,6 +497,8 @@ s8 LoadSystemHacks(bool load_nand)
 			_addOrRejectHack(new_hack);

```

Comment:
Why would I need to do that? I'm always adding all hacks, independantly of their require / provide status, and do the enabling / disabling using the states_hash flag array.

Reply:
because that function is meant to filter out all faulty hacks. 
if a sub wants to be added to the list, but doesn't have a master then its faulty. period.

your logic is in the wrong spot. the loading should do the logic, not afterwards.

Reply:
Okay, I'll move the check to that function.

Reply:
That doesn't really work. When `_addOrRejectHack` is called, we are in the middle of loading hacks. When I encounter a sub hack, I have no idea if its master hack is missing (and thus the hack would be faulty), or if its master hack just occurs later in the file (and thus the hack would be perfectly valid). 

Or would you want to enforce that master hacks always have to appear before their sub hacks in the hacks_hash.ini file?

Reply:
yes, i was intending that a master should be before the sub hacks in the file. its kinda stupid to define a child object before defining a master object...

Reply:
Okay, then I'll add that check tomorrow. And I'll probably add some kind of documentation for the hacks_hash format, explaining all the parameters and restrictions.

Reply:
the only restrictions i can think of atm is master before sub, and a possible lenght limitation. but thats it, no?

Reply:
Yeah, that should be it. It's not that much, but it still might be a good idea to have proper explainations of hash vs offset, provide/require, and so on, so people don't have to guess by looking at existing hacks.